service/builder/pipelines_builder.go

```diff
@@ -318,3 +355,20 @@ func createMetricsProcessor(
 	metricsConverter := converter.NewOCToInternalMetricsConverter(nextConsumer.(consumer.MetricsConsumer))
 	return factoryOld.CreateMetricsProcessor(logger, metricsConverter, cfg)
 }
+
+// createLogProcessor creates a log processor using given factory and next consumer.
+func createLogProcessor(
+	factoryBase component.ProcessorFactoryBase,
+	logger *zap.Logger,
+	cfg configmodels.Processor,
+	nextConsumer consumer.LogConsumer,
+) (component.LogProcessor, error) {
+	if factory, ok := factoryBase.(component.LogProcessorFactory); ok {
+		creationParams := component.ProcessorCreateParams{Logger: logger}
+		ctx := context.Background()
+		return factory.CreateLogProcessor(ctx, creationParams, cfg, nextConsumer)
+	}
+
+	return nil, fmt.Errorf("processor %q does support data type %q",
```

Comment:
Same, use the standard error?

Reply:
I think it is better to keep this more detailed and user friendly error message.